kong/db/schema/typedefs.lua

```diff
@@ -446,15 +449,12 @@ local function validate_path_with_regexes(path)
     return ok, err, err_code
   end
 
-  -- We can't take an ok from validate_path as a success just yet,
-  -- because the router is currently more strict than RFC 3986 for
-  -- non-regex paths:
-  if ngx.re.find(path, [[^[a-zA-Z0-9\.\-_~/%]*$]]) then
+  if path:sub(1, 1) ~= "~" then
```

Comment:
Could this be made backwards compatible by extending this with [#9022](https://github.com/Kong/kong/pull/9022) (keeping the heuristic alive for 3.0, but deprecated)?

```suggestion
  if path:sub(1, 1) ~= "~" and is_not_regex(path) then
```